pkg/util/csi/util.go

```diff
@@ -30,3 +33,48 @@ const (
 func ShouldSkipAction(actionName string) bool {
 	return !features.IsEnabled(velerov1api.CSIFeatureFlag) && strings.Contains(actionName, csiPluginNamePrefix)
 }
+
+func ToSystemAffinity(loadAffinities []*types.LoadAffinity) *corev1.Affinity {
+	if len(loadAffinities) == 0 {
+		return nil
+	}
+
+	result := new(corev1.Affinity)
+	result.NodeAffinity = new(corev1.NodeAffinity)
+	result.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution = new(corev1.NodeSelector)
+	result.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms =
+		make([]corev1.NodeSelectorTerm, 0)
+
+	for _, loadAffinity := range loadAffinities {
+		requirements := []corev1.NodeSelectorRequirement{}
```

Comment:
When all the elements in result in `len(requirements) == 0`, the allocated `result.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution` and `result.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms` are not used but will be passed to the pod creation, which is somehow risky as we don't know how the scheduler interpret them and it is also a unnecessary memory allocation.

Therefore, we can move `requirements := []corev1.NodeSelectorRequirement{}` out of the `for` loop, and then alloc the necessary fields only when after we have the complete `requirements` and it is not empty.